cpu/sam0_common/periph/eth.c

```diff
@@ -225,6 +226,11 @@ int sam0_eth_send(const struct iolist *iolist)
         return -ENOTSUP;
     }
 
+    /* wait until the current buffer is no longer in use */
+    while (!(tx_curr->status & DESC_TX_STATUS_USED)) {
+        thread_yield();
+    }
+
```

Comment:
Isn't too much a slow down here ? Perhaps we could do this only if there is no more buffer available instead.

Reply:
I assumed the buffers would be used sequentially. So we'd only get here if all buffers are in use.

Reply:
Oh right, I miss the placement within the function.